kyuubi-server/src/main/scala/org/apache/kyuubi/server/KyuubiMySQLFrontendService.scala

```diff
@@ -0,0 +1,126 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.server
+
+import java.net.{InetAddress, InetSocketAddress}
+import java.util.concurrent.{ThreadPoolExecutor, TimeUnit}
+
+import io.netty.bootstrap.ServerBootstrap
+import io.netty.buffer.PooledByteBufAllocator
+import io.netty.channel.{ChannelFuture, ChannelInitializer, ChannelOption}
+import io.netty.channel.nio.NioEventLoopGroup
+import io.netty.channel.socket.SocketChannel
+import io.netty.channel.socket.nio.NioServerSocketChannel
+import io.netty.handler.logging.{LoggingHandler, LogLevel}
+
+import org.apache.kyuubi.{KyuubiException, Logging, Utils}
+import org.apache.kyuubi.config.KyuubiConf
+import org.apache.kyuubi.config.KyuubiConf._
+import org.apache.kyuubi.service.{AbstractFrontendService, Serverable, Service}
+import org.apache.kyuubi.util.{NamedThreadFactory, ThreadUtils}
+
+/**
+ * A frontend service implement MySQL protocol.
+ */
+class KyuubiMySQLFrontendService(override val serverable: Serverable)
+  extends AbstractFrontendService("MySQLFrontendService") with Logging {
+
+  private var execPool: ThreadPoolExecutor = _
+
+  var serverAddr: InetAddress = _
+  var port: Int = _
+
+  private var bootstrap: ServerBootstrap = _
+  private var bindFuture: ChannelFuture = _
+
+  @volatile protected var isStarted = false
+
+  override def initialize(conf: KyuubiConf): Unit = synchronized {
+    val poolSize = conf.get(FRONTEND_MYSQL_EXEC_POOL_SIZE)
+    val waitQueueSize = conf.get(FRONTEND_MYSQL_EXEC_WAIT_QUEUE_SIZE)
+    val keepAliveMs = conf.get(FRONTEND_MYSQL_EXEC_KEEPALIVE_TIME)
+    execPool = ThreadUtils.newDaemonQueuedThreadPool(
```

Comment:
can we follow the behavior of ThriftBinaryFrontendService to use ExecutorPoolCaptureOom

Reply:
Done